frontend/views/containers/chatroom/chatroom.js

```diff
@@ -16,117 +55,139 @@ const chatroom = {
     mediaIsPhone.onchange = (e) => { this.config.isPhone = e.matches }
-  },
-  computed: {
-    currentConversation (): Object {
-      // NOTE - This should be $store responsability but for now
-      // I've used the $route just for static mocked layout purposes
-      if (this.$route.params.currentConversation) {
-        return this.$route.params.currentConversation
-      }
-
-      let currentId = null
-      let shouldRefresh = true
 
-      switch (this.$route.name) {
-        case 'Messages':
-          // Open by default the first conversation without unread messages
-          for (let i = 0, l = individualMessagesSorted.length; i < l; i++) {
-            if (users[individualMessagesSorted[i]].unreadCount === 0) {
-              currentId = individualMessagesSorted[i]
-              break
-            }
-          }
-          shouldRefresh = false
-          // fall through
-        case 'MessagesConversation':
-          currentId = currentId || this.findUserId()
-          this.redirectChat('MessagesConversation', users[currentId].name, chatTypes.INDIVIDUAL, currentId, shouldRefresh)
-          break
-        case 'GroupChat': {
-          // Open by default the lounge channel id
-          currentId = 'c0'
-          shouldRefresh = false
-          const chatName = groupA.channels[currentId].name
-          const chatType = chatTypes.GROUP
-
-          this.redirectChat('GroupChatConversation', chatName, chatType, currentId, shouldRefresh)
-          break
-        }
-        case 'GroupChatConversation': {
-          const groupId = currentId || this.findGroupId()
-          currentId = groupId || this.findUserId()
-          const newChatName = groupId ? groupA.channels[currentId].name : users[currentId].name
-          const chatType = groupId ? chatTypes.GROUP : chatTypes.INDIVIDUAL
-
-          this.redirectChat('GroupChatConversation', newChatName, chatType, currentId, shouldRefresh)
-          break
-        }
-        default:
-          break
+    /**
+     * TODO
+     * show toast or dialog that the chatRoomId in URL is not incorrect or that is not-yet-joined chatRoomId
+    **/
+    const { chatRoomId } = this.$route.params
+    if (!chatRoomId) {
+      this.ephemeral.loadedSummary = null
+      this.redirectChat('GroupChatConversation')
+    } else if (!this.isJoinedChatRoom(chatRoomId)) {
+      if (!this.isPublicChatRoom(chatRoomId)) {
+        this.ephemeral.loadedSummary = null
+        this.redirectChat('GroupChatConversation')
+      } else {
+        this.loadSummary()
       }
-
-      return this.$route.params.currentConversation || {}
+    } else {
+      this.refreshTitle()
+    }
+  },
+  computed: {
+    currentChatRoomState (): Object {
+      return this.$store.getters['currentChatRoomState']
+    },
+    currentChatRoomUsers (): Object {
+      return this.$store.getters['chatRoomUsers']
+    },
+    currentChatRoomId (): string {
+      return this.$store.state['currentChatRoomId']
+    },
```

Comment:
Isn't this what [`...mapGetters`](https://vuex.vuejs.org/guide/getters.html#the-mapgetters-helper) is for?

Reply:
Done using `mapGetters`. There was a type annotation error when using `mapGatters` or `mapState`, but fixed it.